lib/Conversion/ImportVerilog/Expression.cpp

```diff
@@ -114,6 +114,10 @@ struct ExprVisitor {
     auto postValue =
         isInc ? builder.create<moore::AddOp>(loc, preValue, one).getResult()
               : builder.create<moore::SubOp>(loc, preValue, one).getResult();
+    if (isPost) {
+      preValue = builder.create<moore::VariableOp>(loc, preValue.getType(),
+                                                   "preValue", arg);
+    }
```

Comment:
Thanks for catching this issue! I don't think you need a separate variable though. You should be able to just swap the order of the assignment and the add/sub op to implement the pre- and post-increment flavors. The variable op will not do what you expect: its initial value is only assigned once at the beginning of the simulation, and won't be updated on subsequent clock ticks. It also introduces a named value into the simulation which doesn't correspond to any named value in the source language, which we should avoid.

Reply:
You'll probably want to fix the following broken IR:
```mlir
 %2 = moore.constant 1 : !moore.int
 %3 = moore.add %b, %2 : !moore.int
 moore.mir.bpassign %b, %3 : !moore.int
 moore.mir.bpassign %a, %b : !moore.int
```
by moving the assignment:
```mlir
 moore.mir.bpassign %a, %b : !moore.int
 %2 = moore.constant 1 : !moore.int
 %3 = moore.add %b, %2 : !moore.int
 moore.mir.bpassign %b, %3 : !moore.int
```